pyne/pnnl/materials_compendium.py

```diff
@@ -0,0 +1,411 @@
+    """
+
+    WeightFraction_whole: float
+    NonIsotopic: bool
+    Element: str
+    WeightFraction: float
+    AtomicMass: float
+    ZAID: str
+    AtomFraction: float
+    AtomDensity_whole: float
+    AtomFraction_whole: float
+    id: str
+    Isotopes: List[Isotope]
+    AtomDensity: float
+    AtomicMass_whole: float
+    Abundances: str
+
+    @staticmethod
+    def from_dict(obj: Any) -> "Element":
+        """
+        Create an Element object from a dictionary representation.
+
+        Parameters:
+            obj (dict): A dictionary containing attributes of the Element.
+
+        Returns:
+            Element: A new Element object with attributes initialized using the values from the dictionary.
+        """
+        _WeightFraction_whole = float(obj.get("WeightFraction_whole"))
+        _NonIsotopic = bool(obj.get("NonIsotopic"))
+        _Element = str(obj.get("Element"))
+        _WeightFraction = float(obj.get("WeightFraction"))
+        _AtomicMass = float(obj.get("AtomicMass"))
+        _ZAID = str(obj.get("ZAID"))
+        _AtomFraction = float(obj.get("AtomFraction"))
+        _AtomDensity_whole = float(obj.get("AtomDensity_whole"))
+        _AtomFraction_whole = float(obj.get("AtomFraction_whole"))
+        _id = str(obj.get("id"))
+        _Isotopes = [Isotope.from_dict(y) for y in obj.get("Isotopes")]
+        _AtomDensity = float(obj.get("AtomDensity"))
+        _AtomicMass_whole = float(obj.get("AtomicMass_whole"))
+        _Abundances = str(obj.get("Abundances"))
+        return Element(
+            _WeightFraction_whole,
+            _NonIsotopic,
+            _Element,
+            _WeightFraction,
+            _AtomicMass,
+            _ZAID,
+            _AtomFraction,
+            _AtomDensity_whole,
+            _AtomFraction_whole,
+            _id,
+            _Isotopes,
+            _AtomDensity,
+            _AtomicMass_whole,
+            _Abundances,
+        )
+
+
+@dataclass
+class Mol:
+    """
+    Represents a molecular entity with various attributes related to its properties.
+
+    Attributes:
+        Mols (int): The number of moles of the molecular entity.
+        Element (str): The name or identifier of the element in the molecular entity.
+        Isotope (str): The name or identifier of the isotope in the molecular entity.
+    """
+
+    Mols: int
+    Element: str
+    Isotope: str
+
+    @staticmethod
+    def from_dict(obj: Any) -> "Mol":
+        """
+        Create a Mol object from a dictionary representation.
+
+        Parameters:
+            obj (dict): A dictionary containing attributes of the Mol.
+
+        Returns:
+            Mol: A new Mol object with attributes initialized using the values from the dictionary.
+        """
+        _Mols = int(obj.get("Mols"))
+        _Element = str(obj.get("Element"))
+        _Isotope = str(obj.get("Isotope"))
+        return Mol(_Mols, _Element, _Isotope)
+
+
+@dataclass
+class Datum:
```

Comment:
Have you thought about this being a `Material` object at this point with all of the metadata attached in the infrastructure we already have for that?

Reply:
Yes, it seems like the `Datum` class could potentially be represented as a `Material` object with additional metadata. But,  the `Material` class already has attributes like `mass`, `density`, `atoms_per_molecule`, and metadata to store arbitrary information. On the other hand, the `Datum` class appears to represent more general data entries. As you have suggested to make a separate algorithm to process PNNL data, I have created this `Datum` class to parse `data: [{...}]`.